lib/rspec/mocks/any_instance/recorder.rb

```diff
@@ -190,9 +191,9 @@ def restore_original_method!(method_name)
           return unless @klass.instance_method(method_name).owner == @klass
 
           alias_method_name = build_alias_method_name(method_name)
-          @klass.class_exec do
+          @klass.class_exec(@backed_up_method_owner) do |backed_up_method_owner|
             remove_method method_name
-            alias_method method_name, alias_method_name
+            alias_method method_name, alias_method_name if backed_up_method_owner[method_name.to_sym] && backed_up_method_owner[method_name.to_sym] == self
```

Comment:
The first part of the `if` conditional seems unnecessary.  It only plays a role if it evaluates to `false` and that can only happen if it returns `false` or `nil`.  The second part checks the return value against `self`, which I don't believe can be `false` or `nil` -- so I think the first part can be removed entirely.  Does that sound right?